### PR TITLE
VAULT-17736 Add HashiCorp contributed label to HC contributed PRs

### PR DESCRIPTION
**Pull request by VioletHynes**
This is the inverse of https://github.com/hashicorp/vault/pull/21600, and was done for reasons explained in the comments of this GHA (forks can't have write permissions). We should be able to find community contributed PRs by adding `-label:hashicorp-contributed-pr` to the search bar.

See also (specifically, the maximum access for forks column is _all_ `read`):
image

I'll have to manually go back and update all open PRs with this label, where appropriate, but this one-time cost seems better than the alternate solution of having a cron job have to re-process every PR every time it runs. I'll make sure everything has the correct label.

### CODE REVIEW

File: .github/workflows/add-hashicorp-contributed-label.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# we can more easily identify community contributed PRs (anything that doesn't
+# have this label).
+# While it might seem like this is the 'reverse' of what we should do, GitHub
+# (rightly) does not allow branches from forks to have write permissions, so
+# making PRs from forks self-label themselves as community-contributed is not
+# possible.
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened, ready_for_review]
```
Comment (miagilepner) [inline .github/workflows/add-hashicorp-contributed-label.yml:13]: do we need the ready_for_review condition? I think the default conditions of opened, synchronize, reopened are sufficient

**Review by miagilepner — APPROVED** [review]

File: .github/workflows/add-hashicorp-contributed-label.yml (hunk 1 above)
Comment (VioletHynes) [reply .github/workflows/add-hashicorp-contributed-label.yml:13]: Good point -- you're right. This runs on draft mode PRs so I think there's no reason to add ready_for_review.

**Comment by maxb** [comment]
@VioletHynes Did you consider using the GitHub pull_request_target event, so that you could securely run triggered by contributor PRs, but in the security context of the target repo, so you can still have write access to add the label?

That would allow you to swap things back around, and label the external contributions explicitly with their own label, instead of having to label the HashiCorp PRs and invert the search.

**Comment by VioletHynes** [comment]
@maxb oh, I missed this event. Damn, this does look perfect for this use case, if I understand correctly. 

I guess it's not too late to add this label to the other PRs and have both, but there's less of a pressing reason to do so now. Still, thanks for the heads up!

**Pull request merged**